### ISSUE TITLE
Use new db - Check/Handle Category ids inconsistencies

### ISSUE THREAD

**Issue by panagiotissp** [labooltd/GameAndroid#39]
@ignacioch  this is the case we discussed in private.

This is an issue from the start.
Ids in the db are not sequential, code always assumed it will be sequential so it crashes if categories 'Sports' and 'Gastronomy' are selected.

We should clarify if this will finally be fixed in the db level.

Otherwise, I will have to write ad-hoc code

Please send an update here when you double check

**Comment by ignacioch** [comment]
So, they are not sequential yes. What I have is this translation to change the index ( diladi to row ) sto equivalent theme/category.


+(int)indexToMappedTheme:(int)indexTheme
--
346 | {
347 | int mappedThemes[10] = { -1 , 1, 2 , 8, 4 , 5 , 9 , 7, 11, 12 }; //# index 0 NEVER USED
348 | return mappedThemes[indexTheme];
349 | }

https://github.com/labooltd/GameOd/blob/6f3db84ce8e0a30bb942be3b1b862065c2416782/GameOd/UtilityClass/Utility.m#L345

This is called from here:

https://github.com/labooltd/GameOd/blob/3e6f0d4ab28767b3b0add40849cf50e09252bc43/GameOd/ViewControllers/GoViewController.m#L185

**Comment by ignacioch** [comment]
Let me know if this is clear.

**Comment by panagiotissp** [comment]
Yes @ignacioch  thank you.
I will use the new db provided (name 'game_content.db') and a mapping like yours inevitably.

Just for confirmation:
- I saw that the difficult column has been dropped.
This means that the 2nd algorithm that uses difficulty will never be used/has been deprecated, right?

- One extra thing since I saw it now that I checked the db-related code: We currently fetch all questions existing in the db from each selected category. Is that aligned with your implementation too?
It works with no problem, and this should be a decision that had been taken long ago ( 2 years before) but just wanted to be 1000% sure that we are still aligned.
